// Copyright 2016 The LUCI Authors. All rights reserved.
// Use of this source code is governed under the Apache License, Version 2.0
// that can be found in the LICENSE file.

// Recompile with libprotoc 2.5.0:
// protoc -I recipe_engine recipe_engine/package.proto --python_out=recipe_engine

syntax = "proto2";

package recipe_engine;

message DepSpec {
  optional string project_id = 1;
  optional string url = 2;
  optional string branch = 3;
  optional string revision = 4;

  // Treat a subtree of a repo as a whole repo unto itself.
  optional string path_override = 5;

  enum RepoType {
    GIT = 0;
    GITILES = 1;
  }
  optional RepoType repo_type = 6 [default = GIT];
}

message Package {
  optional int32 api_version = 1;  // Version 1
  optional string project_id = 2;
  optional string recipes_path = 3;
  repeated DepSpec deps = 4;
}
